code/game/machinery/shuttle/turbine.dm

```diff
@@ -108,49 +124,56 @@
 		return
 
 	if(default_change_direction_wrench(user, I))
-		turbine = null
-		inturf = get_step(src, dir)
-		locate_machinery()
 		if(turbine)
 			to_chat(user, "<span class='notice'>Turbine connected.</span>")
 			set_machine_stat(machine_stat & ~BROKEN)
 		else
 			to_chat(user, "<span class='alert'>Turbine not connected.</span>")
-			obj_break()
 		return
 
 	default_deconstruction_crowbar(I)
 
-/obj/machinery/power/compressor/process()
-	if(!starter)
-		return
-	if(!turbine || (turbine.machine_stat & BROKEN))
-		starter = FALSE
-	if(machine_stat & BROKEN || panel_open)
-		starter = FALSE
-		return
-	cut_overlays()
+//update when moved or changing direction
+/obj/machinery/power/compressor/setDir(newdir)
+	. = ..()
+	locate_machinery()
 
-	rpm = 0.9* rpm + 0.1 * rpmtarget
+/obj/machinery/power/compressor/Move(atom/newloc, direct, glide_size_override)
+	. = ..()
+	locate_machinery()
 
-	// It's a simplified version taking only 1/10 of the moles from the turf nearby. It should be later changed into a better version
-	// above todo 7 years and counting
+/obj/machinery/power/compressor/process(delta_time)
+	return
 
-	inturf.transfer_air_ratio(gas_contained, 0.1)
+/obj/machinery/power/compressor/process_atmos(delta_time)
+	// RPM function to include compression friction - be advised that too low/high of a compfriction value can make things screwy
+	rpm -= 1
+	rpm = (0.9 * rpm) + (0.1 * rpmtarget)
+	rpm = min(rpm, (COMPFRICTION*efficiency)/2)
+	rpm = max(0, rpm - (rpm**2)/(COMPFRICTION*efficiency))
```

Comment:
Not necessarily requesting a change here but ensure `efficiency` isn't ever equal to 0